src/evict.c

```diff
@@ -685,9 +685,10 @@ int performEvictions(void) {
             mem_freed += delta;
             server.stat_evictedkeys++;
             signalModifiedKey(NULL,db,keyobj);
-            notifyKeyspaceEvent(NOTIFY_EVICTED, "evicted",
-                keyobj, db->id);
             propagateDeletion(db,keyobj,server.lazyfree_lazy_eviction);
+            notifyKeyspaceEvent(NOTIFY_EVICTED, "evicted",keyobj,db->id);
+            /* Propagate notification commandds and the del. */
+            propagatePendingCommands();
```

Comment:
i see now we propagate after each operation (which is generally better i guess) - i think we can completely delete `server.propagate_no_multi`